src/audio/speech_to_text/stt_calculator.cc

```diff
@@ -82,42 +99,55 @@ class SttCalculator : public CalculatorBase {
         auto pipe = it->second;
 
         auto payload = cc->Inputs().Tag(INPUT_TAG_NAME).Get<ovms::HttpPayload>();
+        auto endpoint = getEndpoint(payload.uri);
+        if(endpoint == Endpoint::UNSUPPORTED){
+            return absl::InvalidArgumentError(absl::StrCat("Unsupported URI: ", payload.uri));
+        }
 
-        std::unique_ptr<std::string> output;
-        if (absl::StartsWith(payload.uri, "/v3/audio/transcriptions")) {
-            if (payload.multipartParser->hasParseError())
-                return absl::InvalidArgumentError("Failed to parse multipart data");
+        if (payload.multipartParser->hasParseError())
+            return absl::InvalidArgumentError("Failed to parse multipart data");
 
-            std::string_view stream = payload.multipartParser->getFileContentByFieldName("stream");
-            if (!stream.empty()) {
-                return absl::InvalidArgumentError("streaming is not supported");
+        std::string_view stream = payload.multipartParser->getFileContentByFieldName("stream");
+        if (!stream.empty()) {
+            return absl::InvalidArgumentError("streaming is not supported");
+        }
+        std::string_view file = payload.multipartParser->getFileContentByFieldName("file");
+        if (file.empty()) {
+            return absl::InvalidArgumentError(absl::StrCat("File parsing fails"));
+        }
+
+        std::vector<float> rawSpeech;
+        try {
+            if (isWavBuffer(std::string(file))) {
+                SPDLOG_DEBUG("Received file format: wav");
+                rawSpeech = readWav(file);
+            } else {
+                rawSpeech = readMp3(file);
+                SPDLOG_DEBUG("Received file format: mp3");
```

Comment:
```suggestion
                SPDLOG_DEBUG("Received file format: mp3");
                rawSpeech = readMp3(file);
```

in case readMp3 throws we can get debug log earlier, same as in isWavBuffer

Reply:
if readMp3 throws it means that it may not be mp3 format :p

Reply:
I see now